assembl/static2/js/app/pages/idea.jsx

```diff
@@ -426,10 +438,15 @@ const SwitchViewWithContext = props => (
   </DebateContext.Consumer>
 );
 
+const semanticAnalysisForThematicQuery: OperationComponent<SemanticAnalysisForThematicQuery, IdeaQueryVariables, Props> = graphql(
+  SemanticAnalysisForThematicQuery,
+  {
+    props: ({ data }) => data
+  }
+);
+
 export default compose(
-  connect(state => ({
-    lang: state.i18n.locale
-  })),
+  connect(mapStateToProps),
```

Comment:
lang is needed for the IdeaQuery

Reply:
this is how mapStateToProps is defined above in this file:
```
const mapStateToProps = state => ({
  contentLocaleMapping: state.contentLocale,
  timeline: state.timeline,
  defaultContentLocaleMapping: state.defaultContentLocaleMapping,
  lang: state.i18n.locale,
  debateData: state.debate.debateData,
  isHarvesting: state.context.isHarvesting
});
```
Lang is already included

Reply:
Indeed, but why did you replace it? We only needs lang prop here. Having other props in this mapStateToProps before IdeaQuery will just trigger unneeded rerender.

Reply:
I'll fix that but from my point of view we shouldn't have many mapStateToProps in a component. If there is a risk of having unneeded re-rendering, that means the structure itself is not right.